streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3141,8 +3141,8 @@ object ZStream extends Serializable {
   }
 
   final class AccessMPartiallyApplied[R](private val dummy: Boolean = true) extends AnyVal {
-    def apply[E, A](f: R => ZStream[R, Nothing, A]): ZStream[R, Nothing, A] =
-      ZStream.unwrap(ZIO.access(f))
+    def apply[E, A](f: R => ZIO[R, E, A]): ZStream[R, E, A] =
```

Comment:
I think we have to decide what `M` means. If we mean `M` as the "host monad", it makes sense to lock this to `ZStream`", but if we intend it to be "ZIO monad", then it makes more sense to change to `ZIO` as in this pull request.

In any case, the generalization on the error channel is very useful! Probably an oversight in the preceding version.

If all the `mapM` etc. functions on` ZStream` are expecting `ZIO`, then maybe we should change this one to `ZIO`. We could always add `accessStream` for the stream variant, if we decided it was too cumbersome to use the combinators.

Reply:
I think that makes sense. I updated accordingly.